tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -165,5 +167,84 @@ public void AddResponseHeadersToSpanAndCleanup()
         _httpTransport.DisposeAdditiveContext();
     }
 
+    internal static Dictionary<string, object> ExtractCookiesFromRequest(HttpRequest request)
+    {
+        var cookies = RequestDataHelper.GetCookies(request);
+        var cookiesDic = new Dictionary<string, object>();
+
+        if (cookies != null)
+        {
+            for (var i = 0; i < cookies.Count; i++)
+            {
+#if NETCOREAPP || NETSTANDARD
+                var cookie = cookies.ElementAt(i);
+                var keyForDictionary = cookie.Key ?? string.Empty;
+#else
+                var cookie = cookies[i];
+                var keyForDictionary = cookie.Name ?? string.Empty;
+#endif
+                if (!cookiesDic.TryGetValue(keyForDictionary, out var value))
+                {
+                    cookiesDic.Add(keyForDictionary, cookie.Value ?? string.Empty);
```

Comment:
I know it was like this before, but wondering if we really want to send cookies with a null key and/or null value to the waf 🤔

Reply:
Yes, I think that we should exclude cookies with null keys or values. I have updated the code.